ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -554,6 +554,124 @@ public func concatMap<T, U, E>(transform: T -> SignalProducer<U, E>)(producer: S
     return producer |> map(transform) |> concat
 }
 
+/// Returns a producer that forwards values from the latest producer sent on
+/// `producer`, ignoring values sent on previous inner producers.
+///
+/// An error sent on `producer` or the latest inner producer will be sent on the
+/// returned producer.
+///
+/// The returned producer completes when `producer` and the latest inner 
+/// producer have both completed.
+public func latest<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> SignalProducer<T, E> {
+	return SignalProducer<T, E> { sink, disposable in
+		producer.startWithSignal { outerSignal, outerDisposable in
+			disposable.addDisposable(outerDisposable)
+
+			let latestInnerDisposable = SerialDisposable()
+			disposable.addDisposable(latestInnerDisposable)
+			
+			let state = Atomic(LatestState<T, E>.initial)
+			let updateState = { (action: LatestState<T, E> -> LatestState<T, E>) -> () in
+				state.modify(action)
+				if state.value.isComplete {
+					sendCompleted(sink)
+				}
+			}
+			
+			outerSignal.observe(
+				next: { innerProducer in
+					innerProducer.startWithSignal { innerSignal, innerDisposable in
+						latestInnerDisposable.innerDisposable = innerDisposable
+						state.value = state.value.addInnerSignal(innerSignal)
+						
+						innerSignal.observe(SinkOf { event in
+							switch event {
+							case .Completed:
+								updateState { $0.completeInnerSignal(innerSignal) }
+							default:
+								state.withValue { value -> () in
+									if value.isIncompleteLatestInnerSignal(innerSignal) {
```

Comment:
@kastiglione @neilpa Based on https://github.com/ReactiveCocoa/ReactiveCocoa/pull/1759#discussion_r24790861, I think this check is necessary to guarantee an event is never forwarded from a non-latest inner signal. Agree?